{integer}..unresolved_ident suggests field access

### Code

```Rust
fn main() {
    let width = 10;
    // ...
    for _ in 0..w {
        
    }
}
```

### Current output

```
error[E0425]: cannot find value `w` in this scope
 --> src/main.rs:4:17
  |
4 |     for _ in 0..w {
  |                 ^ not found in this scope
  |
help: you might have meant to write `.` instead of `..`
  |
4 -     for _ in 0..w {
4 +     for _ in 0.w {
  |
```

### Desired output

```
error[E0425]: cannot find value `w` in this scope
 --> src/main.rs:4:17
  |
4 |     for _ in 0..w {
  |                 ^ not found in this scope
  |
```

### Rationale and extra context

Since integer literals do not have fields, it's odd to suggest field access.
It just confuses the user, rather than helping.

### Other cases

_No response_

### Rust Version

```
rustc 1.83.0-nightly (6f4ae0f34 2024-10-08)
binary: rustc
commit-hash: 6f4ae0f34503601e54680a137c1db0b81b56cc3d
commit-date: 2024-10-08
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.1
```

### Anything else?

_No response_

--- a/compiler/rustc_resolve/src/late.rs
+++ b/compiler/rustc_resolve/src/late.rs
@@ -4011,6 +4011,7 @@ impl<'a, 'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
                 let instead = res.is_some();
                 let suggestion = if let Some((start, end)) = this.diag_metadata.in_range
                     && path[0].ident.span.lo() == end.span.lo()
+                    && !matches!(start.kind, ExprKind::Lit(_))
                 {
                     let mut sugg = ".";
                     let mut span = start.span.between(end.span);

--- a/tests/ui/range/misleading-field-access-hint.rs
+++ b/tests/ui/range/misleading-field-access-hint.rs
@@ -1,0 +1,8 @@
+// Check if rustc still displays the misleading hint to write `.` instead of `..`
+fn main() {
+    let width = 10;
+    // ...
+    for _ in 0..w {
+        //~^ ERROR cannot find value `w`
+    }
+}

--- a/tests/ui/range/misleading-field-access-hint.stderr
+++ b/tests/ui/range/misleading-field-access-hint.stderr
@@ -1,0 +1,9 @@
+error[E0425]: cannot find value `w` in this scope
+  --> $DIR/misleading-field-access-hint.rs:5:17
+   |
+LL |     for _ in 0..w {
+   |                 ^ not found in this scope
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0425`.

